specification/eventgrid/Azure.Messaging.EventGrid/main.tsp

```diff
@@ -0,0 +1,236 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-autorest";
+import "@azure-tools/typespec-azure-core";
+
+enum ServiceApiVersions {
+  v2023_06_01_preview: "2023-06-01-preview"
+}
+
+@useAuth(
+  ApiKeyAuth<ApiKeyLocation.header, "SharedAccessKey"> | OAuth2Auth<[{
+    type: OAuth2FlowType.implicit,
+    authorizationUrl: "https://login.microsoftonline.com/common/oauth2/v2.0/authorize",
+    scopes: ["https://eventgrid.azure.net/.default"],
+  }]>
+)
+
+@service({
+  title: "AzureMessagingEventGridClient",
+  version: "2023-06-01-preview"
+})
+
+//
+// Supported operations.
+//
+// POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}:publish?api-version={apiVersion}}
+// POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:receive?api-Version={apiVersion}&timeout=60&maxEvents={maxEvents}
+// POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:acknowledge?api-Version={apiVersion}
+// POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:release?api-version={apiVersion}
+//
+
+@server(
+  "{endpoint}",
+  "The host name of the topic",
+  {
+    @doc("The host name of the namespace, e.g. namespaceName1.westus-1.eventgrid.azure.net")
+    endpoint: url,
+  }
+)
+
+@doc("Azure Messaging EventGrid Client")
+@versioned(ServiceApiVersions)
+namespace AzureMessagingEventGrid {
+  using TypeSpec.Http;
+  using TypeSpec.Rest;
+  using TypeSpec.Versioning;
+  using Azure.Core;
+  using Azure.Core.Foundations;
+
+  @doc("Properties of an event published to an Azure Messaging EventGrid Namespace topic using the CloudEvent 1.0 Schema.")
+  model CloudEventEvent {
+    @doc("An identifier for the event. The combination of id and source must be unique for each distinct event.")
+    id: string;
+
+    @doc("Identifies the context in which an event happened. The combination of id and source must be unique for each distinct event.")
+    source: string;
+
+    @doc("Event data specific to the event type.")
+    data?: object;
+
+    @doc("Event data specific to the event type, encoded as a base64 string.")
+    data_base64?: bytes;
```

Comment:
Because this is following the outside spec of [CloudEvent](https://github.com/cloudevents/spec/blob/main/cloudevents/formats/json-format.md#:~:text=CloudEvents%20is%20a%20standardized%20and%20protocol-agnostic%20definition%20of,Notation%20%28JSON%29%20Data%20Interchange%20Format%20%28%20RFC8259%20%29.), it requires these parameters to be lowercase